dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetInfoSerializer.java

```diff
@@ -1,28 +1,38 @@
 package com.dotmarketing.portlets.htmlpageasset.business.render.page;
 
+import com.dotmarketing.portlets.contentlet.transform.ContentletToMapTransformer;
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.JsonSerializer;
 import com.fasterxml.jackson.databind.SerializerProvider;
-import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedMap.Builder;
+import com.google.common.collect.Ordering;
 
 import java.io.IOException;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.Map;
 
+import static com.google.common.collect.Ordering.*;
+
 /**
  * Json Serializer of {@link HTMLPageAssetInfoSerializer}
  */
 public class HTMLPageAssetInfoSerializer extends JsonSerializer<HTMLPageAssetInfo> {
     @Override
-    public void serialize(HTMLPageAssetInfo htmlPageAssetInfo, JsonGenerator jsonGenerator,
-                          SerializerProvider serializerProvider) throws IOException {
+    public void serialize(final HTMLPageAssetInfo htmlPageAssetInfo, final JsonGenerator jsonGenerator,
+                          final SerializerProvider serializerProvider) throws IOException {
+
+        final ContentletToMapTransformer transformer = new ContentletToMapTransformer(htmlPageAssetInfo.getPage());
+        final Map<String, Object> pageContentletMap  = transformer.toMaps().stream().findFirst().orElse(Collections.EMPTY_MAP);
```

Comment:
test?

Reply:
this should be already cover by the current test

Reply:
I already have created a page resource test by postman